config/compilerFlags.cmake

```diff
@@ -8,13 +8,7 @@ if ( MINGW OR UNIX ) # MINGW, Linux, APPLE, CYGWIN
     set (CMAKE_CXX_FLAGS_DEBUG      "-g3 -gstrict-dwarf -O0 -DDEBUG")
 
     if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -Woverloaded-virtual -W")
-
-        if ( CYGWIN OR (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0))
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98") # to support snprintf
-        else()
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98")
-        endif()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -Woverloaded-virtual -W -std=c++11")

```

Comment:
The best way to require C++11 with CMake is by setting the following global CMake variables:

```
set(CMAKE_CXX_STANDARD 11)
set(CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```
This will translate to the specific compiler flags for different compilers. I suggest to remove the `-std=c++11` flag here and use those variables.

Reply:
Didn't know that. I'll include that, that is a lot better than hard coding it via `-std=c++11`. Thanks for the tip!